SQLite/Core/ConnectionPool.swift

```diff
@@ -0,0 +1,177 @@
+//
+// SQLite.swift
+// https://github.com/stephencelis/SQLite.swift
+// Copyright © 2014-2015 Stephen Celis.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+import Dispatch
+import CSQLite
+
+
+private let vfsName = "unix-excl"
+
+
+/// Connection pool delegate
+public protocol ConnectionPoolDelegate {
```

Comment:
The protocol informs listeners that a new connection has been added. Remember that each connection is distinct. So  things like `createFunction` must happen against each added connection; not just the pool itself.

This is the method FMDatabasePool used. The thought had crossed my mind to *capture* calls like `createFunction` and perform them automatically as connections are created. That solution is a long term one that will require a bit of thinking because I don't yet know the scope of what could fall into that category.  The delegate is the obvious first step.